### DISCUSSION TITLE
Space Logistics Simulation - Domain Objects

### DISCUSSION THREAD

**Discussion by mokun** [mars-sim/mars-sim#301] [Ideas]
We currently have a `Resupply` Tool for tracking the incoming landing of supplies sent from Earth.

This MIT/JPL paper called [A Flexible Architecture and Object-Oriented Model for Space Logistics Simulation](https://www.academia.edu/18941864/A_Flexible_Architecture_and_Object_Oriented_Model_for_Space_Logistics_Simulation?auto=download&email_work_card=download-paper) proposed an Object Oriented approach for implementing how the mission logistics of the space mission between nodes in space and was done in MatLab.

Do you guys have any new ideas you would like to see in mars-sim regarding for modeling the space traffic / logistics missions between Earth, Moon and Mars in mars-sim ?

Any merit in using this as a reference document ?

**Comment by Urwumpe** [comment]
Well, the paper does not reinvent the wheel if you read it from a business administration perspective. It also seems to address less space logistic specialties than I would have included - like the relation between launch date and payload mass, and how you can use this for optimizing launch schedules to resupply a base.

The problems they mention about directly working with the database are no surprise at all - that is what OLAP is all about.

I think, it is a good reinforcement to some ideas - you could sure use a flexible OLAP architecture behind the simulation to aggregate and predict the consumption of certain goods. Maybe even just some existing hybrid OLAP implementation for Java.

In other places, it feels like it does not really care about the OLD problems: How do you calculate how much of a certain good was consumed? There are multiple ways known in BA, all with their advantages and disadvantages to real-world problems.

**Comment by mokun** [comment]
@bevans2000 

I want to sketch out more relationship between how the launch mass is related to the launch window and resupply mission.

Anything you see worthy in this pdf paper ?

**Reply by Urwumpe** [reply]
> I want to sketch out more relationship between how the launch mass is related to the launch window and resupply mission.

I would rather recommend some basic spaceflight text books there, for example about the basic relation between payload mass ratio and Delta-V. 

For our logistics purposes, we could assume that it gets launched to LEO for "free" and only the final boost into Mars transfer orbit is calculated by your logistics subroutine. This is essentially just about the basic Tsiolovski equation then.

For calculating the Delta-V requirement for different launch windows, I would recommend Lamberts algorithm. 

https://en.wikipedia.org/wiki/Lambert%27s_problem

You could also take the C++ implementation of PyKEP as reference about how to implement this.

**Comment by mokun** [comment]
>  the basic relation between payload mass ratio and Delta-V.

Right. The mass definitely affects the delta v, thus impacting the arrival time.

> For calculating the Delta-V requirement for different launch windows, I would recommend Lamberts algorithm.

I see. Found pyKEP at https://esa.github.io/pykep/. Will take look later.
